src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -1129,6 +1267,69 @@ public void recordServiceCheckRun(final ServiceCheck sc) {
         send(toStatsDString(sc));
     }
 
+    /**
+     * Updates and returns tags completed with the entityID tag if needed.
+     *
+     * @param tags the current constant tags array
+     *
+     * @param entityID the entityID string provided by argument
+     *
+     * @return array of tags
+     */
+    private String[] updateTagsWithEntityID(String[] tags, String entityID) {
+        // Support "dd.internal.entity_id" internal tag.
+        if (entityID == null) {
+            // if the entityID parameter is null, default to the environment variable
+            entityID = System.getenv(DD_ENTITY_ID_ENV_VAR);
+        }
+        if (entityID != null) {
```

Comment:
Should we also check it's not set to an empty string?

Reply:
I updated the test for detecting empty and also avoid `whitespace` value.